cashctrl_ledger/ledger.py

```diff
@@ -53,17 +53,110 @@ def _single_account_balance():
+                raise ValueError(f"{error_message} '{value}' is duplicated.")
+            return records['id'].item()
+
+        currency_data = self._client.get("currency/list.json")['data']
+        currency_id = get_single_record(pd.DataFrame(currency_data), 'text', currency, "There is no such currency")
+
+        tax_id = None
+        if vat_code:
+            tax_data = self._client.list_tax_rates()
+            tax_id = get_single_record(pd.DataFrame(tax_data), 'text', vat_code, "There is no VAT code")
+
+        category_id = None
+        if group:
+            category_data = self._client.list_categories('account')
+            category_id = get_single_record(pd.DataFrame(category_data), 'path', group, "There is no such group")
+
+        payload = {
+            "number": account,
+            "currencyId": currency_id,
+            "name": text,
+            "taxId": tax_id,
+            "categoryId": category_id,
+        }
+
+        self._client.post("account/create.json", data=payload)
+
+    def update_account(self, account: str, currency: str, text: str, vat_code: str | None = None, group: str | None = None):
         """
-        Not implemented yet
+        Updates an existing account in the remote CashCtrl instance.
+
+        Parameters:
+            account (str): The account number or identifier to be added.
+            currency (str): The currency associated with the account.
+            text (str): Additional text or description associated with the account.
+            vat_code (str, optional): The VAT code to be applied to the account, if any.
+            group (str, optional): The category group to which the account belongs, if any.
         """
-        raise NotImplementedError
+
+        accounts = self._client.list_accounts()
+        account_map = accounts.set_index('number')['id'].to_dict()
+        if account not in account_map:
+            raise ValueError(f"Account '{account}' does not exist.")
+
+        def get_single_record(df, column, value, error_message):
+            records = df[df[column] == value]
+            if len(records) != 1:
+                if len(records) < 1:
+                    raise ValueError(f"{error_message}: '{value}'.")
+                raise ValueError(f"{error_message} '{value}' is duplicated.")
+            return records['id'].item()
+
+        currency_data = self._client.get("currency/list.json")['data']
+        currency_id = get_single_record(pd.DataFrame(currency_data), 'text', currency, "There is no such currency")
+
+        tax_id = None
+        if vat_code:
+            tax_data = self._client.list_tax_rates()
+            tax_id = get_single_record(pd.DataFrame(tax_data), 'text', vat_code, "There is no VAT code")
+
+        category_id = None
+        if group:
+            category_data = self._client.list_categories('account')
+            category_id = get_single_record(pd.DataFrame(category_data), 'path', group, "There is no such group")
```

Comment:
Please replace all occurrences of `get_single_record`with the pattern used for `account_map`.

I am not convinced by the use of `get_single_record`. The interface is not simple and clear, because the function does not fulfil a single purpose. For the benefit of code clarity, I usually avoid this type of glue-code and try to find a clear and succinct way to write out the functionality instead. In our case, we can use the same pattern as with account_map.

Let me compare the two patterns as separate code snippets:
I argue that the first snippet is easier to read.
```
        tax_id = None
        if vat_code:
            tax_data = self._client.list_tax_rates()
            tax_map = tax_data.set_index('text')['id'].to_dict()
            if vat_code not in tax_map:
                raise ValueError(f"VAT code '{vat_code}' does not exist.")
```
```
        def get_single_record(df, column, value, error_message):
            records = df[df[column] == value]
            if len(records) != 1:
                if len(records) < 1:
                    raise ValueError(f"{error_message}: '{value}'.")
                raise ValueError(f"{error_message} '{value}' is duplicated.")
            return records['id'].item()
            
        tax_id = None
        if vat_code:
            tax_data = self._client.list_tax_rates()
            tax_id = get_single_record(pd.DataFrame(tax_data), 'text', vat_code, "There is no VAT code")
```